### PR TITLE
[OTE_SDK_TESTS] Finalizing test_f_measure.py

### PR DESCRIPTION
**Pull request by saltykox**
•	Fixed input dataset for Fmeasure class tests
•	Added scenarios with incorrect dataset for test_f_measure_get_performance

Task: CVS-74378

### CODE REVIEW

**Comment by Ilya-Krylov** [comment]
17:35:12  isort (ote_sdk)..........................................................Passed
17:35:12  isort (ote_cli|external).................................................Passed
17:35:15  black (ote_sdk|ote_cli)..................................................Passed
17:35:16  black (rest).............................................................Passed
17:35:18  flake8...................................................................Passed
17:35:19  prettier.................................................................Passed
17:35:23  mypy (ote_sdk)...........................................................Failed
17:35:23  - hook id: mypy
17:35:23  - exit code: 1
17:35:23  
17:35:23  ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py:1793: error: List item 0 has incompatible type "Optional[CurveMetric]"; expected "CurveMetric"  [list-item]
17:35:23  ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py:1801: error: List item 0 has incompatible type "Optional[ScoreMetric]"; expected "Union[ScoreMetric, CountMetric, InfoMetric, DateMetric, DurationMetric]"  [list-item]
17:35:23  ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py:1807: error: List item 0 has incompatible type "Optional[CurveMetric]"; expected "CurveMetric"  [list-item]
17:35:23  ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py:1815: error: List item 0 has incompatible type "Optional[ScoreMetric]"; expected "Union[ScoreMetric, CountMetric, InfoMetric, DateMetric, DurationMetric]"  [list-item]
17:35:23  Found 4 errors in 1 file (checked 172 source files)
17:35:23  
17:35:24  mypy (ote_cli)...........................................................Passed
17:35:25  mypy (external)..........................................................Passed
17:36:12  pylint...................................................................Passed
17:36:12  [Pipeline] }
17:36:12  ERROR: script returned exit code 1

**Comment by saltykox** [comment]
run ote_sdk tests

**Comment by saltykox** [comment]
run ote_sdk tests

File: ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py [hunk 1]
```diff
@@ -1769,6 +1779,71 @@ def check_performance(performance, expected_score, expected_metrics):
...
+                        palette=ColorPalette.LABEL,
+                        visualization_type=VisualizationType.RADIAL_BAR,
+                    ),
+                )
+            ]
+
+        def generate_expected_optional_dashboard_metric_groups(
+            actual_f_measure: FMeasure,
+        ):
+            per_confidence = CurveMetric(
```
Comment (Ilya-Krylov) [inline ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py]: can not it be just a copy of `actual_f_measure.f_measure_per_confidence` ?

Comment (saltykox) [reply ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py]: It was in commit [https://github.com/openvinotoolkit/training_extensions/pull/845/commits/f1acdf954027fd9c481a3e0adb364e1fed730857](url)
but mypy linter is failed due to [list-item] errors [http://213.221.44.203/job/ote/job/precommit/job/ote_sdk_tests_runner/361/](url) . Should I rollback changes to this commit state and add “# type: ignore” comment to lines caused this errors?

Comment (Ilya-Krylov) [reply ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py]: from typing import cast
        ...
        metrics=[cast(CurveMetric, actual_f_measure.f_measure_per_confidence)]

Comment (saltykox) [reply ote_sdk/ote_sdk/tests/usecases/evaluation/test_f_measure.py]: Thanks, changed the way to generate list of expected dashboard_metrics of Performance object returned by get_performance method

**Comment by saltykox** [comment]
run ote_sdk tests

**Pull request merged**